### PR TITLE
RHIROS-36: add Resource Optimization card on dashboard

### PR DESCRIPTION
**Pull request by PreetiW**
### Details

Added card in the Insights Dashboard to show details related to Resource Optimization service.

Jira Card: https://issues.redhat.com/browse/RHIROS-36

### UI screenshots

**Use Case: Nothing is setup** 
ros_not_configured

**Use Case: Systems with suggestions + waiting for data**
ros_suggestion_waiting_data

Sample JSON output of `/api/ros/v1/is_configured` API

```
{
        "success": true,  
        "code": "SYSTEMSEXIST", 
        "count": 2, 
        "systems_stats": {
                "waiting_for_data": 0, 
                 "with_suggestions": 2
         }
}
```
 
**Note:** 
1) ~~Since we need to update the back-end API is_configured to send few more details related to ROS, the changes will be soon available and hence I have added `fixme` for same. Please feel free to review the rest of the code and let me know any comments/suggestions~~ **The /ros/v1/is_configured API is available in CI/QA environments as of today.**

2) Resource Optimization service is currently only available till beta envrionment, We are planning to go to GA soon and hence these changes will work only till `prod-beta` envrionment at the moment. I will update the Dashboard team about Resource Optimization GA status. Let me know if I need to do anything from my side for same.

### CODE REVIEW

File: src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                                            <Flex
+                                                direction={{ default: 'column' }}
+                                                spaceItems={{ default: 'spaceItemsNone' }}
+                                                alignItems={{ default: 'alignItemsCenter' }}>
+                                                {/* fixme: Need to update it to use values from backend once the API is updated  */}
+                                                <span className='pf-u-font-size-2xl pf-u-color-100 pf-u-font-weight-bold'>
+                                                2
+                                                </span>
+                                                <span className='pf-u-font-size-sm'>
+                                                Systems waiting for data
```
Comment (karelhala) [inline src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js:60]: Can we add this string to translations as well?

File: src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                                                spaceItems={{ default: 'spaceItemsNone' }}
+                                                alignItems={{ default: 'alignItemsCenter' }}>
+                                                {/* fixme: Need to update it to use values from backend once the API is updated  */}
+                                                <span className='pf-u-font-size-2xl pf-u-color-100 pf-u-font-weight-bold'>
+                                                10
+                                                </span>
+                                                <span className='pf-u-font-size-sm'>
+                                                Systems with suggestions
+                                                </span>
+                                            </Flex>
```
Comment (karelhala) [inline src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js:74]: This one as well please.

File: src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                                                direction={{ default: 'column' }}
+                                                spaceItems={{ default: 'spaceItemsNone' }}
+                                                alignItems={{ default: 'alignItemsCenter' }}>
+                                                {/* fixme: Need to update it to use values from backend once the API is updated  */}
+                                                <span className='pf-u-font-size-2xl pf-u-color-100 pf-u-font-weight-bold'>
+                                                15
+                                                </span>
+                                                <span className='pf-u-font-size-sm'>
+                                                Total systems
+                                                </span>
```
Comment (karelhala) [inline src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js:85]: And this one.

File: src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+                                        onClick={e => navigateTo(e, `${UI_BASE}/ros`)}
+                                        href={ `${UI_BASE}/ros`}>
+                                        {intl.formatMessage(messages.rosCardConfigureSystemsCTA)}
+                                    </Button>
+                                </Flex>
+                        )
+
+                    }
+                    {rosIsConfiguredFetchStatus === 'pending' && <Loading />}
+                    {rosIsConfiguredFetchStatus === 'rejected' && <FailState appName='Resource optimization' />}
```
Comment (karelhala) [inline src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js:115]: I would translate the `Resource optimization` as well, but that's probably a product name, so let's not translate it.

**Review by karelhala — COMMENTED** [review]
Looking good! Just a few translations missing, but overall looks solid.

**Review by Hyperkid123 — COMMENTED** [review]
As @karelhala pointed out, other than the missing string, everything looks good.

File: src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js (hunk 1 above)
Comment (PreetiW) [reply src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js]: agree, updated 💯

File: src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js (hunk 2 above)
Comment (PreetiW) [reply src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js]: agree, updated 100

File: src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js (hunk 3 above)
Comment (PreetiW) [reply src/SmartComponents/ResourceOptimization/ResourceOptimizationCard.js]: agree, updated 100

File: locales/translations.json [hunk 5]
```diff
@@ -131,5 +131,13 @@
...
-  "zssampleTitleR": "CVSS Base Score"
+  "zssampleTitleR": "CVSS Base Score",
+  "resourceOptimizationCardHeader": "Resource optimization",
+  "rosSystemsNotConfiguredMessage": "No systems configured yet? Get started with resource optimization.",
+  "rosCardConfigureSystemsCTA": "Get started with configuration",
+  "rosSystemsGenericMessage": "Red Hat recommends to assess and monitor cloud usage and optimization on these systems.",
+  "rosCardViewSystemsCTA": "View systems",
+  "waitingForData": "Systems waiting for data",
+  "systemsWithSuggestions": "Systems with suggestions",
+  "totalSystems": "Total systems"
```
Comment (PreetiW) [inline locales/translations.json:142]: @karelhala @Hyperkid123 just a small question for my understanding, I observed we are using `data.json` in the AppEntry.js for transaltion so I am not sure how we using translation.json file. Can you please share some insights on this? Thanks in advance.

**Review by PreetiW — COMMENTED** [review]
Thank you @karelhala and @Hyperkid123 for reviewing my PR 💯 
I have updated the code with missing transaltions and logic to use actual API data. I wanted to inform you that the `is_configure` API is now available in the CI and QA environment so we can merge this PR for those environment if everything looks good and I will update the description with the sample json output from the API.

**Comment by PreetiW** [comment]
@karelhala @Hyperkid123 also can you let me what I will need to do to update the team that the ROS API is available in the different environment as we are not GA yet (planning to go GA soon) I want to know more about the release cycle of Insights Dashboard as we can only push changes till QA environment as of today. Thanks in advance!

**Review by karelhala — APPROVED** [review]
Looking good!

**Pull request merged**